### ISSUE TITLE
Random data generator

### ISSUE THREAD

**Issue by JakeLin** [IBAnimatable/IBAnimatable#78]
Random name, random address, random title, random text? 

Depends on [Plugin for prototyping only](https://github.com/JakeLin/IBAnimatable/issues/77)

**Comment by tbaranes** [comment]
Random color, random gradient? (we can even go further and play with [Chameleon ](https://github.com/unifiedh/Chameleon-Swift) to have a nice result)

**Comment by JakeLin** [comment]
Yes, I don't think random color or gradient is useful, I rather choose them in predefined color/gradient set. random name, avatar and image will be very helpful for prototyping since <del>all of us are</del> I am lazy.

**Comment by tbaranes** [comment]
Depending, it would be nice to randomise a gradient / color relative to a defined background. For example, when prototyping, we could just specify a color, and have everything around nice.

Nevertheless, I agree with you that is less useful than names, address...

**Comment by JakeLin** [comment]
Well, that app must be very smart to create relevant colors. it is something like this https://color.adobe.com/

**Comment by tbaranes** [comment]
Exactly! That's why I was saying it shouldn't be a priority :laughing:

**Comment by lexrus** [comment]
https://github.com/timuric/Content-generator-sketch-plugin
http://www.invisionapp.com/craft
It would be awesome if we can do the same things in storyboard.
How about we start from an easy one, names label. Implement as a protocol extension in a CocoaPods sub spec called 'Prototyping'.

**Comment by JakeLin** [comment]
@lexrus that's why I am thinking of, there are some dataset when we work on it like http://uinames.com/  and http://www.radfaces.com/

May create `PrototypableLable: AnimatableLable` to add on those properties, but I am not sure can we build another Pod within the same framework? Or we can have a Prototyping framework based on IBAnimatable framework?

**Comment by tbaranes** [comment]
We can create a [subspec](https://guides.cocoapods.org/syntax/podspec.html#group_subspecs) if we want to keep everything in IBAnimatable. That means we can specify in our Podfile which spec we wants: 

```
pod 'IBAnimatable'
pod 'IBAnimatable/Prototyping'
```

However, I'm not sure that it's a good idea to keep everything in `IBAnimatable` since a prototyping framework is a big job apart.

**Comment by phimage** [comment]
Hi guys, I have some code which can help

I make a proof of concept project
Here a video https://www.dropbox.com/s/pj61i4t2m86cl8p/Ibfakery.mov?dl=0
https://github.com/phimage/IBFakery

In user defined runtime attributes
- to the left we assign the type of data
- to the right, the field where the value must be set

I use swift framework https://github.com/vadymmarkov/Fakery for my fake data
And I make a little code for random colors (but other frameworks could be used like https://github.com/nvzqz/RandomKit)
A KVC compliant framework for fake and random data will be more easy to integrate

**Comment by JakeLin** [comment]
@phimage it is very cool. I see the code is quite small but super powerful 💪

**Comment by tbaranes** [comment]
Should we move that issue in a dedicated project and why not adding that project in IBAnimatable organization (if that's ok with you @phimage)? 🙃

**Comment by phimage** [comment]
Yes, it's ok with me
The idea come after reading this issue
